lua/lspconfig/groovyls.lua

```diff
@@ -12,7 +12,7 @@ configs[name] = {
       bin_name,
     },
     filetypes = { 'groovy' },
-    root_dir = util.find_git_ancestor or vim.loop.os_homedir(),
+    root_dir = util.root_pattern 'Jenkinsfile' or util.find_git_ancestor,
```

Comment:
This won't work, it will just always assign the _function_ `util.root_pattern` to `root_dir`.

Reply:
You're basically doing this:

```console
> function foo() end
> function bar() end
> a = foo or bar
> print(a == foo)
true
> print(b == bar)
false
```

Reply:
Ah, I was afraid of that. (I briefly tested, but with an inconclusive example ;))

Looks like you have to follow this pattern: https://github.com/neovim/nvim-lspconfig/blob/a4deb0c770bed049218f183f6a0de249c52d1d69/lua/lspconfig/robotframework_ls.lua#L10-L12

Reply:
Ah, I think we can do this then: `util.root_pattern 'Jenkinsfile'(fname) or util.find_git_ancestor(fname)`

Reply:
But wrap it in a function, like for `robotframework_ls`.

Reply:
Ah, that was bad info -- that is only the case for table literals, not string literals. (I opened an issue about that, but no changes were made yet.) Sorry!